R/create_default_parameters.R

```diff
@@ -0,0 +1,584 @@
+  check_valid_input(input = form, valid_options = valid_forms, arg_name = "recruitment form")
+
+  # Create default parameters based on the recruitment form
+  process_default <- switch(form,
+    "BevertonHoltRecruitment" = create_default_BevertonHoltRecruitment(data)
+  )
+  names(process_default) <- paste0(form, ".", names(process_default))
+
+  # Create default distribution parameters based on the distribution type
+  distribution_input <- recruitment[["process_distribution"]]
+  distribution_default <- NULL
+  if (!is.null(distribution_input)) {
+    distribution_default <- switch(distribution_input,
+      "DnormDistribution" = create_default_DnormDistribution(
+        value = 0.1,
+        data = data,
+        input_type = "process"
+      )
+    )
+    names(distribution_default) <- paste0(distribution_input, ".", names(distribution_default))
+  }
+
+  # Combine process and distribution defaults into a single list
+  default <- list(c(process_default, distribution_default))
+  names(default) <- "recruitment"
+  return(default)
+}
+
+#' Update Input Parameters for a FIMS Model
+#'
+#' @description
+#' This function updates the input parameters of a FIMS model.
+#' It allows users to modify specific parameters by providing new values, while
+#' retaining the existing modules information from the current input.
+#' @param current_parameters A list containing the current input parameters, including:
+#'   \describe{
+#'     \item{\code{parameters}:}{A list of parameter inputs.}
+#'     \item{\code{modules}:}{A list of module names used in the model.}
+#'   }
+#' @param modified_parameters A named list representing new parameter values to update.
+#' @rdname create_default_parameters
+#' @return A list containing:
+#'   \item{parameters}{A list of updated parameter inputs that
+#'   includes any modifications made by the user.}
+#'   \item{modules}{The unchanged list of module names from the current
+#'   input.}
+#' @export
+update_parameters <- function(current_parameters, modified_parameters) {
+  # Check if current_parameters is a list with required components
+  if (!is.list(current_parameters) || !all(c("parameters", "modules") %in% names(current_parameters))) {
+    cli::cli_abort("The {.var current_parameters} argument must be a list containing parameters and modules.")
+  }
+
+  # Check if modified_parameters is a named list
+  if (!is.list(modified_parameters) || is.null(names(modified_parameters))) {
+    cli::cli_abort("The {.var modified_parameters} argument must be must be a named list.")
+  }
+
+  # Check if modified_parameters exists in current_parameters
+  missing_input <- setdiff(names(modified_parameters), names(current_parameters[["parameters"]]))
+  if (length(missing_input) > 0) {
+    cli::cli_abort(c(
+      "x" = "The following {length(missing_input)} input list{?s} from
+            {.var modified_parameters} {?is/are} missing from
+            {.var current_parameters}: {paste(missing_input, collapse = ', ')}"
```

Comment:
Inside of `glue::glue()`, which is what is used by {cli} you do not have to put commas in between vector items because it will do it for you. It even puts the and 😄.

Reply:
That's awesome. I'll use glue more often!